service.php

```diff
@@ -317,6 +317,21 @@ function token_select_box( $name, $create = false ) {
 		$tokens = $this->get_tokens();
 		return Keyring_Util::token_select_box( $tokens, $name, $create );
 	}
+
+	/**
+	 * A shortcut to query registered endpoints.  It saves having to re-extract the url and method manually.
+	 *
+	 * @param string $endpoint The registered endpoint (via `->set_endpoint()`) to query.
+	 * @param array $params Additional parameters for the request (a la WP_HTTP)
+	 * @return Keyring_Error|String Either an error or the response data.
+	 */
+	function request_endpoint( $endpoint, $params = array() ) {
```

Comment:
This function doesn't appear to be used anywhere in this changeset. It seems like a decent idea in general, but probably better to introduce it, and convert Services over to using it, all in the same change.